CMakeLists.txt

```diff
@@ -914,6 +921,44 @@ if(EMSCRIPTEN)
         COMPONENT ktx_js
     )
 
+    add_executable( ktx_js_read interface/js_binding/ktx_wrapper.cpp )
+    target_link_libraries( ktx_js_read ktx_read )
+    target_include_directories(
+        ktx_js_read
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/other_include
+        ${CMAKE_CURRENT_SOURCE_DIR}/lib
+        $<TARGET_PROPERTY:ktx_read,INTERFACE_INCLUDE_DIRECTORIES>
+    )
+    target_link_options(
+        ktx_js_read
+    PUBLIC
+        ${KTX_EMC_LINK_FLAGS}
+        "SHELL:-s EXPORT_NAME=LIBKTX_READ"
```

Comment:
I'd like to avoid duplicating these SHELL: items. Can you put them in a CMake variable to be used in both builds?

Reply:
I was trying to deduplicate the definitions of the read-only vs read-write versions, but didn't turn out to be any cleaner. If you really want to, I can do that nonetheless.

Reply:
I found the proper way to do this, setting them as INTERFACE items so they are automatically set for dependencies. See commit 5d13ab5. You'll pick it when merging or rebasing main.